### PR TITLE
Add Button.Group

### PR DESCRIPTION
**Pull request by emplums**
This PR adds a `Button.Group` component for grouped buttons! 🎉

### CODE REVIEW

File: src/Button.js [hunk 1]
```diff
@@ -27,6 +40,21 @@ Button.defaultProps = {
   theme
 }
 
+const Group = ({children, ...rest}) => {
+  const newChildren = React.Children.map(children, child => React.cloneElement(child, {grouped: true}))
```
Comment (colebemis) [inline src/Button.js]: We could avoid using `React.Children` and `React.cloneElement` by having `Group` apply styles its children. Here's a codesandbox example: https://codesandbox.io/s/buttongroup-u2h46

I don't know that approach is any better 🤷‍♂ What do you think?

File: src/ButtonStyles.js [hunk 2]
```diff
@@ -42,25 +42,50 @@ const getButtonStyles = theme => {
...
+          border-bottom-right-radius: ${get('radii.1')(theme)}px;
+        }
+
+        &:focus,
+        &:active,
+        &:hover {
+          border-right-width: 1px;
+
+          + .grouped {
+            border-left-width: 0;
```
Comment (colebemis) [inline src/ButtonStyles.js]: This is neat! Adding back the right border on hover makes the button group feel very polished 👍

File: src/ButtonGroup.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    :focus,
+    :active {
+      z-index: 1;
+    }
+  }
+`
+ButtonGroup.defaultProps = {
+  display: 'inline-block'
```
Comment (colebemis) [inline src/ButtonGroup.js]: If you set `display: inline-block` in the styled components definition, does the system prop not work?

Comment (emplums) [reply src/ButtonGroup.js]: Yeah, it overrides it because the styled-system prop gets added first

**Pull request merged**